src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java

```diff
@@ -64,11 +66,13 @@
  * <li>
  * Property {@code excludeClassesRegexps} - Specify user-configured regular
  * expressions to ignore classes.
+ * Type is {@code java.lang.String[]}.
```

Comment:
even internal type is List<Patten>
https://github.com/checkstyle/checkstyle/blob/7d4cb0da6268d4ee7bc6ab94728b114316eefc3e/src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java#L83

javadoc will have real type of setter, in during xdoc generation we will do some extra javadoc line to provide `Validation Type is {@code Pattern[]}` probably.

Reply:
Should we provide `Validation Type` also in metadata generation?